From 31f775cf2aa5c00355e1987d77ba540fd929bb73 Mon Sep 17 00:00:00 2001
From: Artur Wieczorek <artwik@wp.pl>
Date: Fri, 3 Apr 2015 21:47:13 +0200
Subject: [PATCH 2/4] Use wxMilliClock_t type variables to store values
 returned by wxGetLocalTimeMillis() function in wxComboCtrl.

wxMilliClock_t is always mapped to the proper base type and hence can be used even if wxLongLong datatype is not defined (when wxUSE_LONGLONG is disabled).
---
 include/wx/combo.h      | 2 +-
 include/wx/msw/combo.h  | 2 +-
 src/common/combocmn.cpp | 2 +-
 src/msw/combo.cpp       | 6 +++++-
 4 files changed, 8 insertions(+), 4 deletions(-)

diff --git a/include/wx/combo.h b/include/wx/combo.h
index 57fbcd8..86b08a6 100644
--- a/include/wx/combo.h
+++ b/include/wx/combo.h
@@ -647,7 +647,7 @@ protected:
     // used to prevent immediate re-popupping in case closed popup
     // by clicking on the combo control (needed because of inconsistent
     // transient implementation across platforms).
-    wxLongLong              m_timeCanAcceptClick;
+    wxMilliClock_t          m_timeCanAcceptClick;
 
     // how much popup should expand to the left/right of the control
     wxCoord                 m_extLeft;
diff --git a/include/wx/msw/combo.h b/include/wx/msw/combo.h
index 50dc6eb..9ab1b30 100644
--- a/include/wx/msw/combo.h
+++ b/include/wx/msw/combo.h
@@ -100,7 +100,7 @@ private:
 
 #if wxUSE_COMBOCTRL_POPUP_ANIMATION
     // Popup animation related
-    wxLongLong  m_animStart;
+    wxMilliClock_t m_animStart;
     wxTimer     m_animTimer;
     wxRect      m_animRect;
     int         m_animFlags;
diff --git a/src/common/combocmn.cpp b/src/common/combocmn.cpp
index f346ea9..8fb1354 100644
--- a/src/common/combocmn.cpp
+++ b/src/common/combocmn.cpp
@@ -1918,7 +1918,7 @@ bool wxComboCtrlBase::HandleButtonMouseEvent( wxMouseEvent& event,
 bool wxComboCtrlBase::PreprocessMouseEvent( wxMouseEvent& event,
                                             int WXUNUSED(flags) )
 {
-    wxLongLong t = ::wxGetLocalTimeMillis();
+    wxMilliClock_t t = ::wxGetLocalTimeMillis();
     int evtType = event.GetEventType();
 
 #if USES_WXPOPUPWINDOW || USES_GENERICTLW
diff --git a/src/msw/combo.cpp b/src/msw/combo.cpp
index 3f9b3ac..d10aa0e 100644
--- a/src/msw/combo.cpp
+++ b/src/msw/combo.cpp
@@ -731,10 +731,14 @@ void wxComboCtrl::DoTimerEvent()
     }
     else
     {
-        wxLongLong t = ::wxGetLocalTimeMillis();
+        wxMilliClock_t t = ::wxGetLocalTimeMillis();
         const wxRect& rect = m_animRect;
 
+#if wxUSE_LONGLONG
         int pos = (int) (t-m_animStart).GetLo();
+#else
+        int pos = (int) (t-m_animStart);
+#endif
         if ( pos < COMBOBOX_ANIMATION_DURATION )
         {
             int height = rect.height;
-- 
1.9.5.msysgit.0

